derive/src/util.rs

```diff
@@ -1,3 +1,216 @@
-pub fn path_eq(path: &syn::Path, s: &str) -> bool {
+use super::Diagnostic;
+use std::collections::HashMap;
+use syn::{Attribute, AttributeArgs, Ident, Lit, Meta, NestedMeta, Path};
+
+pub fn path_eq(path: &Path, s: &str) -> bool {
     path.get_ident().map_or(false, |id| id == s)
 }
+
+pub fn def_to_name(
+    ident: &Ident,
+    attr_name: &'static str,
+    attr: AttributeArgs,
+) -> Result<String, Diagnostic> {
+    let mut name = None;
+    for attr in attr {
+        if let NestedMeta::Meta(meta) = attr {
+            if let Meta::NameValue(name_value) = meta {
+                if path_eq(&name_value.path, "name") {
+                    if let Lit::Str(s) = name_value.lit {
```

Comment:
Should we do any `magic` name formatting here, or is that sorted out on the name collected here later on?

Reply:
Oh, nevermind, this is just copied from `pyclass.rs`